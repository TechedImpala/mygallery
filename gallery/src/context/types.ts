import { ApplicationStateEnum } from "../types";

export interface IAppProviderState {
  activeState: ApplicationStateEnum;
  setActiveState: React.Dispatch<
    React.SetStateAction<ApplicationStateEnum>
  > | null;
  activeTopic: string;
  setActiveTopic: React.Dispatch<React.SetStateAction<string>> | null;
  menuVisible: boolean;
  setMenuVisibility: React.Dispatch<React.SetStateAction<boolean>> | null;
}
