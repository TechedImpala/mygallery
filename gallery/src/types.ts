export enum ApplicationStateEnum {
  GRID = "GRID",
  MENU = "MENU",
}

export interface IUrls {
  raw: string;
  full: string;
  regular: string;
  small: string;
  thumb: string;
  small_s3: string;
}
