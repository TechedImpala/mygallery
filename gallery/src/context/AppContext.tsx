import { createContext, useContext, useState } from "react";
import { ApplicationStateEnum } from "../types";
import { IAppProviderState } from "./types";

const AppContext = createContext<IAppProviderState>({
  activeState: ApplicationStateEnum.MENU,
  setActiveState: null,
  activeTopic: "",
  setActiveTopic: null,
  menuVisible: true,
  setMenuVisibility: null,
});

interface IAppProviderProps {
  children: React.ReactNode | React.ReactNode[];
}

const AppProvider = ({ children }: IAppProviderProps) => {
  const [menuVisible, setMenuVisibility] = useState(true);
  const [activeState, setActiveState] = useState<ApplicationStateEnum>(
    ApplicationStateEnum.MENU
  );

  const [activeTopic, setActiveTopic] = useState<string>("");

  const value = {
    activeState,
    setActiveState,
    activeTopic,
    setActiveTopic,
    menuVisible,
    setMenuVisibility,
  };

  return <AppContext.Provider value={value}>{children}</AppContext.Provider>;
};

export const useAppContext = () => useContext(AppContext);

export default AppProvider;
