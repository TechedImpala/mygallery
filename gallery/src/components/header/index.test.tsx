import React from "react";
import { render, screen } from "@testing-library/react";
import Header from "./";
import { useAppContext } from "../../context/AppContext";
import { ApplicationStateEnum } from "../../types";

jest.mock("../../context/AppContext");

describe("Header:", () => {
  test("correct text when menu state is active", () => {
    (useAppContext as jest.Mock).mockReturnValue({
      activeState: ApplicationStateEnum.MENU,
    });
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });

  test("correct text when grid state is active", () => {
    (useAppContext as jest.Mock).mockReturnValue({
      activeState: ApplicationStateEnum.GRID,
    });
    const { container } = render(<Header />);

    expect(container).toMatchSnapshot();
  });
});
