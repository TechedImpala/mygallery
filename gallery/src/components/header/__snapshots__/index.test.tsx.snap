// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header: correct text when grid state is active 1`] = `
<div>
  <header
    class="sc-bczRLJ gIpnIp"
  >
    State: Menu is inactive and grid is active
  </header>
</div>
`;

exports[`Header: correct text when menu state is active 1`] = `
<div>
  <header
    class="sc-bczRLJ gIpnIp"
  >
    State: Menu is active and grid is inactive
  </header>
</div>
`;
